### PR TITLE
fix merge of contacts blanking affiliations

### LINKED ISSUE #2753: Contact Merge: If Primary Affiliation on non-master Contact, clean up is removing Primary Affiliation from Contact record
Steps to repeat:

1. Contact1 w/Primary Affiliation to organization
2. Contact2 with no Primary Affiliation
3. Contact Merge, selecting the Primary Affiliation value from the non-master record

When the clean up runs to delete the duplicate Affiliation record, the correct Affiliation record checked as Primary remains, the duplicate is deleted, however the Primary Affiliation value is cleared from the Contact record incorrectly.

### PR DESCRIPTION
**Pull request by patrick-yan-sf**
Fix a bug where merging of contacts with affiliations is not consistent.

# Critical Changes

# Changes

# Issues Closed
Fixes #2753 

# New Metadata

# Deleted Metadata

### CODE REVIEW

**Comment by patrick-yan-sf** [comment]
**lurch: attach W-020499

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x80000008FH2HAAW" target="_blank">W-020499</a>

File: src/classes/CON_ContactMerge_TEST2.cls [hunk 1]
```diff
@@ -880,4 +817,103 @@ private class CON_ContactMerge_TEST2 {
...
+            List<npe5__Affiliation__c> affiliations = [
+                SELECT Id, npe5__Contact__c, npe5__EndDate__c, npe5__Organization__c, 
+                    npe5__Primary__c, npe5__Role__c, npe5__StartDate__c, npe5__Status__c, npe5__Description__c
+                FROM npe5__Affiliation__c 
+                WHERE npe5__Contact__c IN :contactsToMerge
+            ];
+
+            System.assertEquals(1, affiliations.size(), 'Affiliations are created for the contact');
+            
+            // Test.setCurrentPageReference(new PageReference('Page.CON_ContactMerge'));
```
Comment (RadSFDeveloper) [inline src/classes/CON_ContactMerge_TEST2.cls]: Do we need this comment?

Comment (patrick-yan-sf) [reply src/classes/CON_ContactMerge_TEST2.cls]: No we don't.  I forgot to remove it after I refactored the test function to include it.  I've removed it now. Thanks

**Comment by patrick-yan-sf** [comment]
@rponti-sforg - This is ready for testing

**Pull request merged**